src/myprofile-page/ContactInfo.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Avatar,
+  Card, CardActions, CardContent, IconButton, Input, InputLabel,
+} from '@material-ui/core';
+import { Edit } from '@material-ui/icons';
+import './styles.css';
+
+const ContactInfo = ({ actualUser, onUpdate, isUpdating }) => {
+  const defaultPicture = 'https://www.worldfuturecouncil.org/wp-content/uploads/2020/06/blank-profile-picture-973460_1280-1.png';
+  const [avatarSource, setAvatarSource] = useState(actualUser.avatarUrl);
+  const [isEditable, setIsEditable] = useState(false);
+
+  const fields = [
+    'city', 'country', 'email', 'phone', 'website', 'about',
+  ];
+
+  const renderContent = () => (
+    <Card className="contact-card">
+      <CardContent className="contact-card-content">
+
+        <Avatar
+          className="avatar"
+          src={avatarSource}
+          onError={() => setAvatarSource(defaultPicture)}
+          variant="circular"
+        />
+
+        {fields.map((field) => (
+          <div key={field}>
+            <InputLabel htmlFor={`${field}input`}>
+              {field.toUpperCase()}
+            </InputLabel>
+            <Input
+              fullWidth
+              disableUnderline
+              multiline
+              id={`${field}input`}
+              disabled={!isEditable}
+              name={field}
+              value={actualUser[field]}
+              onChange={(event) => {
+                onUpdate({
+                  ...actualUser,
+                  [event.target.name]: event.target.value,
+                });
+                isUpdating(true);
+              }}
+            />
+            <br />
+            <br />
+          </div>
+        ))}
+
+      </CardContent>
+      <CardActions>
+        <IconButton
+          size="small"
+          onClick={() => setIsEditable(!isEditable)}
```

Comment:
If the next state value depends on the previous value, as in this example, it's better to use the functional version of setState. https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
updated for the `setIsEditable` statements